### PR TITLE
Fix CI and bump actions dependencies to deal with GHA deprecations.

### LINKED ISSUE #82: CI is utterly broken
Noticed in https://github.com/matrix-org/synapse-s3-storage-provider/pull/80

https://github.com/matrix-org/synapse-s3-storage-provider/commit/ffd3fa477321608e57d27644197e721965e0e858 was good

https://github.com/matrix-org/synapse-s3-storage-provider/commit/5a70b9ac8f21b11cafa78f78f5c7571ad8ad8b28 was bad: [integration test failed](https://github.com/matrix-org/synapse-s3-storage-provider/actions/runs/2977969293/jobs/4772286737)

Then in https://github.com/matrix-org/synapse-s3-storage-provider/commit/e762c859ac7910e043f09f45a657ff607c66a599 the unit tests suddenly started failing: https://github.com/matrix-org/synapse-s3-storage-provider/actions/runs/3233934664/jobs/5296573191 . Was unable to reproduce this on my machine

### PR DESCRIPTION
**Pull request by DMRobertson**
Fixes #82.

Drive-by contribution to https://github.com/matrix-org/synapse/issues/14203

### CODE REVIEW

**Comment by DMRobertson** [comment]
This seems to work? But the unit tests shouldn't now be passing---the fix only affects the integration test? Extremely suspicious.

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -34,12 +34,14 @@ jobs:
                 secret_access_key: my_secret_access_key
     - name: Setup minio
       run: |
+        # The version of minio is RELEASE.2022-04-01T03-41-39Z.fips
```
Comment (clokep) [inline .github/workflows/test.yml:37]: Did something change in minio that we're pinning it?

**Review by clokep — APPROVED** [review]

File: .github/workflows/test.yml (hunk 1 above)
Comment (DMRobertson) [reply .github/workflows/test.yml:37]: I haven't dug into this. I tried to pin minio to the version we ought to have been running the last time this test succeeded. It seemed to work. :shrug:

Comment (clokep) [reply .github/workflows/test.yml:37]: OK.

Comment (DMRobertson) [reply .github/workflows/test.yml:37]: I thought about running a bisection, but being honest I didn't want to faff around with GHA and editing yaml.

**Pull request merged**